src/status_im/contexts/wallet/add_account/create_account/events.cljs

```diff
@@ -202,3 +203,38 @@
                                    derived-from-address
                                    %)}]]]}))
 
+(rf/reg-event-fx
+ :wallet/set-seed-phrase
+ (fn [{:keys [db]} [{:keys [seed-phrase on-success]}]]
```

Comment:
There is no reason for this event to have `on-success` since it doesn't fail anywhere and that `on-success` not invoked conditionally.
If something else should happen along with `set-seed-phrase` it can happen without `set-seed-phrase` knowing about it)

Reply:
Also passing `on-success` directly to the `fx` effect doesn't seem well :thinking: 

@seanstrom Maybe this is the kind of problem you wanted to solve  with your effect?

CC:  @ilmotta

Reply:
> Also passing on-success directly to the fx effect doesn't seem well

Yes, this is leaking an internal detail about re-frame. This means the caller of `:wallet/set-seed-phrase` will need to decide the effect this event should use (`:dispatch`, `:dispatch-later`, etc). It works, but it complicates things.

Reply:
I'm not certain, but fwiw I don't think this event will be needed after the code is changed to use the `create-account-from-private-key` effect. I think this event was being used to help generate a seed-phrase to generate an account, but I don't think that's desired since we're attempting to import the private-key and generate an account based on the private-key.

I mention this behaviour in more detail and have some example code linked in this comment: https://github.com/status-im/status-mobile/pull/20423#pullrequestreview-2113455692

Reply:
Yeah, I agree, thanks @seanstrom !

Thanks @ilmotta